preprocessor/src/lib.rs

```diff
@@ -0,0 +1,94 @@
+use whistle_lexer::Lexer;
+use whistle_lexer::LexerError;
+use whistle_lexer::LexerErrorKind;
+use whistle_common::TokenItem;
+use whistle_common::Literal;
+use whistle_common::Range;
+use whistle_common::Keyword;
+use whistle_common::Token;
+
+#[derive(Default, Clone)]
+pub struct Preprocessor {
+    token_list: Vec<Vec<TokenItem>>,
+}
+
+impl Preprocessor {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn process(&mut self, src: &str) -> Result<(), LexerError> {
+        let mut lexer = Lexer::new(src);
+        let mut tokens: Vec<TokenItem> = Vec::new();
+
+        let mut imports: Vec<String> = Vec::new();
+        loop {
+            let item = match lexer.next() {
+                Some(v) => v?,
+                None => break,
+            };
+
+            if item.token == Token::Keyword(Keyword::Import) {
+                let import = match lexer.next() {
+                    Some(v) => v?,
+                    None => return Err(LexerError::new(LexerErrorKind::Eof, item.range))
+                };
+
+                let import_file = match import.token {
+                    Token::Literal(lit) => {
+                        match lit {
+                            Literal::Str(s) => s,
+                            _ => return Err(LexerError::new(LexerErrorKind::ExpectedStringStartDelim, import.range))
+                        }
+                    },
+                    _ => return Err(LexerError::new(LexerErrorKind::ExpectedStringStartDelim, import.range))
+                };
+                
+                imports.push(import_file);
+            } else {
+                tokens.push(item);
+            }
+        }
+        for mut file_name in imports {
+            let file_data =  if file_name.starts_with("http://") || file_name.starts_with("https://") {
+                match reqwest::blocking::get(file_name) {
+                    Ok(v) => match v.text() {
+                        Ok(v) => v,
+                        Err(_) => return Err(LexerError::new(LexerErrorKind::UnexpectedEof, Range { start: 0, end: 0})),
+                    },
+                    Err(_) => return Err(LexerError::new(LexerErrorKind::UnexpectedEof, Range { start: 0, end: 0})),
+                }
+            } else if file_name.starts_with("@") {
+                &file_name.remove(0);
+                match reqwest::blocking::get("https://raw.githubusercontent.com/whistle-lang/std/main/".to_owned()+&file_name) {
```

Comment:
Could we instead have the standard library locally stored in a kindof `WHISTLE_DIR`? Otherwise I suggest we keep the long url for now and in the future set up a whistle registry instead of this magic implied base url.